openedx/core/djangoapps/user_api/helpers.py

```diff
@@ -65,16 +66,19 @@ def _wrapped(*args, **kwargs):
                         LOGGER.warning(msg)
                         raise
 
+                caller = traceback.format_stack(limit=2)[0]
+
                 # Otherwise, log the error and raise the API-specific error
                 msg = (
                     u"An unexpected error occurred when calling '{func_name}' "
-                    u"with arguments '{args}' and keyword arguments '{kwargs}': "
+                    u"with arguments '{args}' and keyword arguments '{kwargs}' from {caller}: "
                     u"{exception}"
                 ).format(
                     func_name=func.func_name,
                     args=args,
                     kwargs=kwargs,
-                    exception=ex.developer_message if hasattr(ex, 'developer_message') else repr(ex)
+                    exception=ex.developer_message if hasattr(ex, 'developer_message') else repr(ex),
+                    caller=caller.strip()
```

Comment:
Nit: Consider adding a trailing comma for cleaner diffs / less developer aggravation.